src/main/java/fi/okm/jod/yksilo/config/suomifi/Saml2LoginConfig.java

```diff
@@ -201,4 +208,18 @@ public void onLogoutSuccess(
       handle(request, response, null);
     }
   }
+
+  @RequiredArgsConstructor
+  static class ProfileDeletionHandler implements LogoutHandler {
+    private final YksiloService yksiloService;
+
+    @Override
+    public void logout(
+        HttpServletRequest request, HttpServletResponse response, Authentication authentication) {
+      if ("true".equals(request.getParameter("deletion"))) {
+        var principal = (JodSaml2Principal) authentication.getPrincipal();
```

Comment:
JodSaml2Principal implements JodUser, so use it instead. That would also allow keeping the service delete signature (JodUser instead if raw id), and would make the handled independent of the authentication method (we might need to implement this for other methods as well).